e2e/main.e2e.spec.ts

```diff
@@ -0,0 +1,81 @@
+import path from 'path';
+import { Application, SpectronClient } from 'spectron';
+import { BrowserWindow } from 'electron';
+import NavBarPage from './pages/navbar';
+import GeneralPage from './pages/general';
+import KubernetesPage from './pages/kubernetes';
+import PortForwardingPage from './pages/portforwarding';
+import ImagesPage from './pages/images';
+import TroubleshootingPage from './pages/troubleshooting';
+const electronPath = require('electron');
+
+jest.setTimeout(1_000_000);
+
+describe('Rancher Desktop', () => {
+  let app:Application;
+  let client: SpectronClient;
+  let browserWindow: BrowserWindow;
+  let navBarPage: NavBarPage;
+  let generalPage: GeneralPage;
+  let kubernetesPage: KubernetesPage;
+  let portForwardingPage: PortForwardingPage;
+  let imagesPage: ImagesPage;
+  let troubleShootingPage: TroubleshootingPage;
+
+  beforeAll(async() => {
+    app = new Application({
+      path:             electronPath as any,
+      args:             [path.join(__dirname, '..')],
+      webdriverOptions: {},
+      env:              { NODE_ENV: 'test' }
```

Comment:
To avoid depending on NODE_ENV which is set in the build script plus to keep things specific to spectron, I have introduced a new env variable called SPECTRON_RUN. Please refer the commit 87acb67.